From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Armbian <info@armbian.com>
Date: Tue, 27 May 2026 00:00:00 +0800
Subject: u-boot: configs: Add fly_h618_defconfig with eMMC support

FLY-H618 has eMMC on MMC2 which needs CONFIG_MMC_SUNXI_SLOT_EXTRA=2
and CONFIG_SUPPORT_EMMC_BOOT. Based on orangepi_zero3_defconfig
with these additions plus HS200 mode for 8-bit eMMC.

---
 configs/fly_h618_defconfig | 38 ++++++++++++++++++++++++++++++++++++++
 1 file changed, 38 insertions(+)
 create mode 100644 configs/fly_h618_defconfig

diff --git a/configs/fly_h618_defconfig b/configs/fly_h618_defconfig
new file mode 100644
index 000000000000..000000000001
--- /dev/null
+++ b/configs/fly_h618_defconfig
@@ -0,0 +1,38 @@
+CONFIG_ARM=y
+CONFIG_ARCH_SUNXI=y
+CONFIG_DEFAULT_DEVICE_TREE="allwinner/sun50i-h618-fly-h618"
+CONFIG_OF_UPSTREAM=y
+CONFIG_SPL=y
+CONFIG_MACH_SUN50I_H616=y
+CONFIG_SUNXI_DRAM_H616_LPDDR4=y
+CONFIG_DRAM_CLK=792
+CONFIG_MMC_SUNXI_SLOT_EXTRA=2
+CONFIG_MMC_HS200_SUPPORT=y
+CONFIG_SPL_MMC_HS200_SUPPORT=y
+CONFIG_R_I2C_ENABLE=y
+# CONFIG_SYS_MALLOC_CLEAR_ON_INIT is not set
+CONFIG_SPL_I2C=y
+CONFIG_SPL_SYS_I2C_LEGACY=y
+CONFIG_SYS_I2C_MVTWSI=y
+CONFIG_SYS_I2C_SLAVE=0x7f
+CONFIG_SYS_I2C_SPEED=400000
+CONFIG_SUN8I_EMAC=y
+CONFIG_SUPPORT_EMMC_BOOT=y
+CONFIG_AXP313_POWER=y
+CONFIG_USB_EHCI_HCD=y
+CONFIG_USB_OHCI_HCD=y
+CONFIG_USB_MUSB_GADGET=y
-- 
2.35.3
